OrbitClientData/ModuleData.cpp

```diff
@@ -8,14 +8,30 @@
 #include "OrbitClientData/FunctionUtils.h"
 #include "absl/synchronization/mutex.h"
 #include "capture_data.pb.h"
+#include "module.pb.h"
 
 using orbit_client_protos::FunctionInfo;
+using orbit_grpc_protos::ModuleInfo;
 
 bool ModuleData::is_loaded() const {
   absl::MutexLock lock(&mutex_);
   return is_loaded_;
 }
 
+void ModuleData::Update(ModuleInfo info) {
+  absl::MutexLock lock(&mutex_);
+
+  CHECK(file_path() == info.file_path());
+
+  // if a property changed, remove functions
```

Comment:
I would check build_id and only if it is empty - check the rest of properties instead. It could be that the file_size is changed but build_id is not (in the case we get another debug section stripped for example without a rebuild).

Reply:
done